### PR TITLE
Show new semester optional subjects

### PR DESCRIPTION
**Pull request by RenzoMinelli**
Add a new page in the menu to display the semester's subjects from Inco.
Currently the yml was made by hand but we could update the scrapper to get that data autoamtically.

### CODE REVIEW

File: app/views/shared/_menu.html.erb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
       <%= drawer_menu_navigation_item('Avance de Carrera', root_path) %>
       <%= drawer_menu_navigation_item('Todas las Materias', all_subjects_path) %>
+      <%= drawer_menu_navigation_item('Materias INCO semestre actual', current_optional_subjects_path) %>
```
Comment (EmilioCristalli) [inline app/views/shared/_menu.html.erb:7]: Should it be `Optativas INCO semestre actual` or something like that?

File: db/data/scraped_optional_subjects.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+- '5914'
```
Comment (EmilioCristalli) [inline db/data/scraped_optional_subjects.yml:2]: probably no need to make them strings (ie: `- 5914`)

File: app/services/yml_loader.rb [hunk 3]
```diff
@@ -42,6 +43,14 @@ def load_subjects
     end
   end
 
+  # rubocop:disable Rails/SkipsModelValidations
+  def load_current_optional_subjects
+    optional_subject_codes = YAML.load_file(Rails.root.join("db/data/scraped_optional_subjects.yml"))
+    Subject.update_all(current_optional_subject: false)
+    Subject.where(code: optional_subject_codes).update_all(current_optional_subject: true)
```
Comment (EmilioCristalli) [inline app/services/yml_loader.rb:50]: should we do this in a transaction?

File: app/views/current_optional_subjects/_optional_subjects_list.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="mdc-deprecated-list">
+  <% subjects.each do |subject| %>
+    <%= link_to subject_path(subject), :class => "mdc-deprecated-list-item" do %>
```
Comment (EmilioCristalli) [inline app/views/current_optional_subjects/_optional_subjects_list.html.erb:3]:
```suggestion
    <%= link_to subject_path(subject), class: "mdc-deprecated-list-item" do %>
```

File: app/views/current_optional_subjects/_optional_subjects_list.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<div class="mdc-deprecated-list">
```
Comment (EmilioCristalli) [inline app/views/current_optional_subjects/_optional_subjects_list.html.erb:1]: can we use `app/views/subjects/_subjects_list.html.erb` instead?
i think it's exactly the same

File: app/views/current_optional_subjects/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  <%= render 'shared/credits_counter' %>
+<% end %>
+
+<% content_for :welcome_banner do %>
+  <% unless current_student.welcome_banner_viewed? %>
+    <%= render 'shared/welcome_banner' %>
+  <% end %>
+<% end %>
+
+<%= render partial: 'optional_subjects_list', locals: { subjects: @subjects } %>
```
Comment (EmilioCristalli) [inline app/views/current_optional_subjects/index.html.erb:11]:
```suggestion
<%= render "subjects/subjects_list", subjects: @subjects %>
```

File: app/views/current_optional_subjects/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<% content_for :credits do %>
+  <%= render 'shared/credits_counter' %>
+<% end %>
+
+<% content_for :welcome_banner do %>
+  <% unless current_student.welcome_banner_viewed? %>
+    <%= render 'shared/welcome_banner' %>
```
Comment (EmilioCristalli) [inline app/views/current_optional_subjects/index.html.erb:7]: i wonder if we should render the banner or not. I think right now we only do it in the home page, right?

File: app/controllers/current_optional_subjects_controller.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+class CurrentOptionalSubjectsController < ApplicationController
+  def index
+    @subjects = Subject.current_semester_optionals.order(:code)
+
+    respond_to do |format|
+      format.html
+    end
```
[the remark this answers is not in the document]
Comment (RenzoMinelli) [reply app/controllers/current_optional_subjects_controller.rb]: Removed

File: app/views/shared/_menu.html.erb (hunk 1 above)
Comment (RenzoMinelli) [reply app/views/shared/_menu.html.erb:7]: Actually some subjects are required subjects not only optional like `Proyecto De Grado`

File: app/services/yml_loader.rb (hunk 3 above)
Comment (RenzoMinelli) [reply app/services/yml_loader.rb]: Updated

File: app/views/current_optional_subjects/_optional_subjects_list.html.erb (hunk 4 above)
Comment (RenzoMinelli) [reply app/views/current_optional_subjects/_optional_subjects_list.html.erb]: Updated

File: app/views/current_optional_subjects/_optional_subjects_list.html.erb (hunk 5 above)
Comment (RenzoMinelli) [reply app/views/current_optional_subjects/_optional_subjects_list.html.erb]: Agreed, removed.

File: app/views/current_optional_subjects/index.html.erb (hunk 6 above)
Comment (RenzoMinelli) [reply app/views/current_optional_subjects/index.html.erb]: Updated

File: app/views/current_optional_subjects/index.html.erb (hunk 7 above)
Comment (RenzoMinelli) [reply app/views/current_optional_subjects/index.html.erb]: Removed

File: db/data/scraped_optional_subjects.yml (hunk 2 above)
Comment (RenzoMinelli) [reply db/data/scraped_optional_subjects.yml]: Now the scrapper generates this file

**Review by santiagorodriguez96 — APPROVED** [review]
LGTM :shipit:

**Pull request merged**